src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php

```diff
@@ -95,21 +100,68 @@ public function parse(array $item) : array
      *
      * @return array
      */
-    private function parseClass(ResourceMetadata $resourceMetadata, string $resourceClass, string $io, array $visited = []) : array
+    private function parseResource(ResourceMetadata $resourceMetadata, string $resourceClass, string $io, array $visited = []) : array
     {
         $visited[] = $resourceClass;
 
-        $attributes = $resourceMetadata->getAttributes();
         $options = [];
-        $data = [];
 
-        if (isset($attributes['normalization_context']['groups'])) {
-            $options['serializer_groups'] = $attributes['normalization_context']['groups'];
-        }
+        $options['serializer_groups'] = $resourceMetadata->getAttribute(
+                'normalization_context',
+                ['groups' => []]
+            )['groups'];
 
-        if (isset($attributes['denormalization_context']['groups'])) {
-            $options['serializer_groups'] = isset($options['serializer_groups']) ? array_merge($options['serializer_groups'], $attributes['denormalization_context']['groups']) : $options['serializer_groups'];
-        }
+        $options['serializer_groups'] = array_merge(
+                $options['serializer_groups'],
+                $resourceMetadata->getAttribute('denormalization_context', ['groups' => []])
+            )['groups'];
+
+        return $this->getPropertyMetadata($resourceMetadata, $resourceClass, $io, $visited, $options);
+    }
+
+    /**
+     * Returns groups of item & collection.
+     *
+     * @param ResourceMetadata $resourceMetadata
+     * @param string           $resourceClass
+     * @param string           $io
+     * @param string[]         $visited
+     *
+     * @return array
+     */
+    private function getGroupsForItemAndCollectionOperation(ResourceMetadata $resourceMetadata, string $resourceClass, string $io, string $operationName, array $visited = []) : array
```

Comment:
This is what i'm doing inside of it. Look at the parse method, I added the operationName because I needed to. In in order to not have the parse method full of code I made this method.